app/helpers/lookup_keys_helper.rb

```diff
@@ -74,35 +74,39 @@ def overridable_lookup_keys(klass, host)
     klass.class_params.override.where(:environment_classes => {:environment_id => host.environment}) + klass.lookup_keys
   end
 
-  def hostgroup_key_with_diagnostic(hostgroup, key)
-    value, origin = hostgroup.inherited_lookup_value key
-    original_value = key.value_before_type_cast value
-    diagnostic_helper = popover('', _("<b>Description:</b> %{desc}<br><b>Type:</b> %{type}<br> <b>Matcher:</b> %{matcher}") % { :desc => key.description, :type => key.key_type, :matcher => origin}, :data => { :placement => 'top' })
-    parameter_value_content("value_#{key.key}", original_value, :popover => diagnostic_helper)
-  end
+  def key_with_diagnostic(obj, key, index, value_hash = nil)
+    if value_hash.nil? # hostgroup
+      value, matcher = obj.inherited_lookup_value key
+    else # host
+      value_for_key = value_hash[key.id] && value_hash[key.id][key.key]
+      value, matcher = value_for_key ? [value_for_key[:value], "#{value_for_key[:element]} (#{value_for_key[:element_name]})"] : [key.default_value, _("Default value")]
+    end
 
-  def host_key_with_diagnostic(host, value_hash, key)
-    value_for_key = value_hash[key.id] && value_hash[key.id][key.key]
-    value, matcher = value_for_key ? [value_for_key[:value], "#{value_for_key[:element]} (#{value_for_key[:element_name]})"] : [key.default_value, _("Default value")]
     original_value = key.value_before_type_cast value
-    no_value = value.nil? && key.lookup_values.find_by_match("fqdn=#{host.fqdn}")
+    lookup_value = obj.lookup_values.detect{|v| v.lookup_key_id == key.id}
+    overriden_value = lookup_value.try(:value)
+    no_value = value.to_s.empty? && overriden_value.to_s.empty?
 
-    if no_value
+    if no_value && value_hash.present?
       if key.required
-        diagnostic_class = 'error'
+        diagnostic_class = 'has-error'
         diagnostic_helper = popover('', _("Required parameter without value.<br/><b>Please override!</b> <br><br><b>Description:</b>: %s") % key.description, :icon => "exclamation-sign")
       else
-        diagnostic_class = 'warning'
+        diagnostic_class = 'has-warning'
         diagnostic_helper = popover('', _("Optional parameter without value.<br/><i>Won\'t be given to Puppet.</i> <br><br><b>Description:</b> %s") % key.description, :icon => "warning-sign")
       end
     else
-      diagnostic_helper = popover('', _("<b>Description:</b> %{desc}<br><b>Type:</b> %{type}<br> <b>Matcher:</b> %{matcher}") % { :desc => key.description, :type => key.key_type, :matcher => matcher}, :data => { :placement => 'top' })
+      diagnostic_helper = popover('', _("<b>Description:</b> %{desc}<br/><b>Type:</b> %{type}<br/> <b>Matcher:</b> %{matcher} <br/><b>Value:</b> %{original_value}") % { :desc => key.description, :type => key.key_type, :matcher => matcher, :original_value => original_value }, :data => { :placement => 'top' }, :title => _("Original value info"))
```

Comment:
Note that the popover title states this is all info related to the original value, do you think this needs more clarificaion?